### PR TITLE
Updating signup and login functions for more secure passwords.

### PR DESCRIPTION
**Pull request by ddritzenhoff**
Every new user will now have his/her password hashed. I also created a migration that will update all of the old users' passwords. This still needs to be run, though.

### CODE REVIEW

**Comment by ddritzenhoff** [comment]
I tested both functions, and they worked. The only thing that I didn't test is the migration file b/c that would mess stuff up if it isn't on master (I think)

File: api/migrations/0004_auto_20210313_0206.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('api', '0003_usercontact'),
+    ]
+
+    operations = [
+        migrations.RunPython(update_passwords)
```
Comment (mattognibene) [inline api/migrations/0004_auto_20210313_0206.py:23]: so will this hash all of the passwords in the db?

**Review by mattognibene — APPROVED** [review]
This is awesome, nice job. One question about the migration, but this should be G2G. Since all of our accounts are just test accounts anyway I have no problem dropping that data to get this change in.

File: api/migrations/0004_auto_20210313_0206.py (hunk 1 above)
Comment (ddritzenhoff) [reply api/migrations/0004_auto_20210313_0206.py:23]: yeah

**Pull request merged**